lib/data/data_sources/remote_data_source/channels_data_source.dart

```diff
@@ -1,42 +1,56 @@
 import 'dart:async';
 
 import 'package:sendbird_sdk/sendbird_sdk.dart';
-import 'package:sendbirdtutorial/Core/string_constants.dart';
+
+import '../../../Core/string_constants.dart';
 
 class ChannelsDataSource extends ChannelEventHandler {
   final SendbirdSdk sendbird;
 
-  StreamController<BaseMessage> _messageStreamController;
-  StreamController<BaseChannel> _channelsStreamController;
+  late StreamController<BaseMessage> _messageStreamController;
+  late StreamController<BaseChannel> _channelsStreamController;
 
-  ChannelsDataSource({this.sendbird}) {
+  ChannelsDataSource({required this.sendbird}) {
     _messageStreamController = StreamController<BaseMessage>();
 
     _channelsStreamController = StreamController<GroupChannel>(
-        onListen: () => sendbird.addChannelEventHandler(StringConstants.channelHandlerKey, this),
-        onCancel: () => sendbird.removeChannelEventHandler(StringConstants.channelHandlerKey));
+        onListen: () => sendbird.addChannelEventHandler(
+            StringConstants.channelHandlerKey, this),
+        onCancel: () => sendbird
+            .removeChannelEventHandler(StringConstants.channelHandlerKey));
   }
 
-  Stream<BaseMessage> get onChannelNewMessage => _messageStreamController.stream;
+  Stream<BaseMessage> get onChannelNewMessage =>
+      _messageStreamController.stream;
 
-  Stream<GroupChannel> get onChannelNewChanged => _channelsStreamController.stream;
+  Stream<BaseChannel> get onChannelNewChanged =>
+      _channelsStreamController.stream;
 
   void closeStream() {
     _messageStreamController.close();
     _channelsStreamController.close();
   }
 
+  // ChannelEventHandler methods
+  @override
+  void onMessageReceived(BaseChannel channel, BaseMessage message) {
+    _messageStreamController.sink.add(message);
+  }
+
   @override
   void onChannelChanged(BaseChannel channel) {
-    GroupChannel groupChannel = channel;
+    BaseChannel groupChannel = channel;
```

Comment:
is this local variable needed? can we use channel directly without defining another variable?